src/app/core/services/transactions.service.ts

```diff
@@ -72,27 +67,35 @@ export class TransactionsService {
     tokens: TokenSymbol[],
     chainId: ChainId
   ): void {
-    let transactionScanner = TransactionsService.TRANSACTION_SCANNERS[chainId];
-    const txUrl = transactionScanner ? `${transactionScanner.url}/${hash}` : '';
     const transactions = this.pendingTransactions$.value.transactions || [];
-    const pendingTransactions: IPendingTransactions = {
-      transactions: [
-        ...transactions,
-        {
-          hash,
-          type,
-          tokens,
-          txUrl,
-          scanner: transactionScanner ? transactionScanner.name : '',
-          chainId,
-        },
-      ],
-    };
-    this.pendingTransactions$.next(pendingTransactions);
-    localStorage.setItem(
-      TransactionsService.PENDING_TX_KEY,
-      JSON.stringify(pendingTransactions)
-    );
+    if (!hash || transactions.find((tx) => tx.hash === hash)) {
+      return;
+    }
+    this.connectorService.providerUserInfo$.pipe(take(1)).subscribe((info) => {
+      const txUrl = info.chainInfo.chain_scanner_base_url
+        ? `${info.chainInfo.chain_scanner_base_url}/tx/${hash}`
+        : '';
+      const pendingTransactions: IPendingTransactions = {
+        transactions: [
+          ...transactions,
+          {
+            hash,
+            type,
+            tokens,
+            txUrl,
+            scanner: info.chainInfo.chain_scanner_name
+              ? info.chainInfo.chain_scanner_name
+              : info.chainInfo.name,
```

Comment:
can you please move this bellow txUrl and create a const called for exemple scannerName ?